dotnet/src/IntegrationTests/Plugins/RepairServiceTests.cs

```diff
@@ -106,7 +108,112 @@ public async Task HttpOperationExceptionIncludeRequestInfoAsync()
+        };
+        result = await plugin["createRepair"].InvokeAsync(kernel, arguments);
+
+        Assert.NotNull(result);
+        Assert.Equal("New repair created", result.ToString());
+
+        // List All Repairs
+        result = await plugin["listRepairs"].InvokeAsync(kernel);
+
+        Assert.NotNull(result);
+        repairs = JsonSerializer.Deserialize<Repair[]>(result.ToString());
+        Assert.True(repairs?.Length > 0);
+        Assert.Equal(count + 2, repairs?.Length);
+    }
+
+    [Fact(Skip = "This test is for manual verification.")]
+    public async Task UseDelegatingHandlerAsync()
+    {
+        // Arrange
+        var kernel = new Kernel();
+        kernel.FunctionInvocationFilters.Add(new CreateRepairFilter());
+        using var stream = System.IO.File.OpenRead("Plugins/repair-service.json");
+
+        using var httpHandler = new HttpClientHandler();
+        using var customHandler = new CustomHandler(httpHandler);
+        using HttpClient httpClient = new(customHandler);
+
+        var plugin = await kernel.ImportPluginFromOpenApiAsync(
+            "RepairService",
+            stream,
+            new OpenAIFunctionExecutionParameters(httpClient) { IgnoreNonCompliantErrors = true, EnableDynamicPayload = false });
+
+        // List All Repairs
+        var result = await plugin["listRepairs"].InvokeAsync(kernel);
+
+        Assert.NotNull(result);
+        var repairs = JsonSerializer.Deserialize<Repair[]>(result.ToString());
+        Assert.True(repairs?.Length > 0);
+        var count = repairs?.Length ?? 0;
+
+        // Create Repair - oil change
+        var arguments = new KernelArguments
+        {
+            ["payload"] = """{ "title": "Engine oil change", "description": "Need to drain the old engine oil and replace it with fresh oil.", "assignedTo": "", "date": "", "image": "" }"""
+        };
+        result = await plugin["createRepair"].InvokeAsync(kernel, arguments);
+
+        Assert.NotNull(result);
+        Assert.Equal("New repair created", result.ToString());
+
+        // Create Repair - brake pads change
+        arguments = new KernelArguments
+        {
+            ["payload"] = """{ "title": "Brake pads change", "description": "Need to replace the brake pads on all wheels.", "assignedTo": "", "date": "", "image": "" }"""
+        };
+        result = await plugin["createRepair"].InvokeAsync(kernel, arguments);
+
+        Assert.NotNull(result);
+        Assert.Equal("New repair created", result.ToString());
+
+        // List All Repairs
+        result = await plugin["listRepairs"].InvokeAsync(kernel);
+
+        Assert.NotNull(result);
+        repairs = JsonSerializer.Deserialize<Repair[]>(result.ToString());
+        Assert.True(repairs?.Length > 0);
+        Assert.Equal(count + 2, repairs?.Length);
+    }
+
+    public sealed class Repair
```

Comment:
Make it private, as this class is being used only within this test class

```suggestion
    private sealed class Repair
```